### PR TITLE
Fix/14060: Bug: Link with multiline text loses markdown when edited

### PR DESCRIPTION
**Pull request by tienifr**
### Fixed Issues
$ https://github.com/Expensify/App/issues/14060

# Tests
1. Login with any account and go to any chat room
2. Send this message:
```
[multiline
text](https://google.com/)
```
3. Edit this message

# QA
1. Login with any account and go to any chat room
2. Send this message:
```
[multiline
text](https://google.com/)
```
3. Edit this message
4. Verify that when editing, the original message should be auto-filled
```
[multiline
text](https://google.com/)
```

### CODE REVIEW

File: lib/ExpensiMark.js [hunk 1]
```diff
@@ -267,7 +267,7 @@ export default class ExpensiMark {
             },
             {
                 name: 'anchor',
-                regex: /<(a)[^><]*href\s*=\s*(['"])(.*?)\2(?:".*?"|'.*?'|[^'"><])*>(.*?)<\/\1>(?![^<]*(<\/pre>|<\/code>))/gi,
+                regex: /<(a)[^><]*href\s*=\s*(['"])(.*?)\2(?:".*?"|'.*?'|[^'"><])*>(.*?)<\/\1>(?![^<]*(<\/pre>|<\/code>))/gis,
```
Comment (mananjadhav) [inline lib/ExpensiMark.js:270]: I recommend we use `\s\S` and it is uniform with our usage.

Comment (mananjadhav) [inline lib/ExpensiMark.js:270]: We generally also add Tests for all the markdown changes. Please check previous PRs or other Tests for reference.

Comment (tienifr) [reply lib/ExpensiMark.js]: Sorry about my confusion

Comment (tienifr) [reply lib/ExpensiMark.js]: Sorry about my confusion

**Review by mananjadhav — APPROVED** [review]

**Comment by mananjadhav** [comment]
@cristipaval All yours. This is the `expensify-common` PR. There'll be a follow up PR to add it to the App.

**Review by Luke9389 — APPROVED** [review]
Looks great, thanks @tienifr.
In the future, feel free to add me to any relevant PRs that I'm assigned as the CME on (https://github.com/Expensify/App/issues/14060)

**Comment by Luke9389** [comment]
I'll leave this open for @cristipaval to review and merge.

**Review by cristipaval — APPROVED** [review]
Thanks @Luke9389 !

**Pull request merged**